orogen_syskit_testsTypes.hpp

```diff
@@ -0,0 +1,18 @@
+#ifndef orogen_syskit_tests_TYPES_HPP
+#define orogen_syskit_tests_TYPES_HPP
+
+namespace orogen_syskit_tests {
+    struct Time {
+        int microseconds = 0;
+        int tv_sec = 0;
+        int tv_usec = 0;
+    };
+
+    struct TestType {
+        /** @meta role logical_time */
+        Time timestamp;
+        int test_value = 0;
+    };
```

Comment:
You need to use base::Time or a type that has an equivalent structure. The logger code is expecting a `int64_t microseconds` (not sure it validates that, but it definitely should).

It would work here on 64 bit machines, but would fail on 32 (since `int` is 32 bit long on these)

Actually, I would suggest validating it in Syskit in the method that selects the field.